src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -104,16 +106,39 @@ function is(x, y) {
 /*eslint-enable no-self-compare*/
 
 function createChainableTypeChecker(validate) {
+  if (__DEV__) {
+    var manualPropTypeCallCache = {};
+  }
   function checkType(
     isRequired,
     props,
     propName,
     componentName,
     location,
-    propFullName
+    propFullName,
+    secret
   ) {
     componentName = componentName || ANONYMOUS;
     propFullName = propFullName || propName;
+    if (__DEV__) {
+      if (
+        secret !== ReactPropTypesSecret &&
+        typeof console !== 'undefined'
+      ) {
+        var cacheKey = `${componentName}:${propName}`;
+        if (!manualPropTypeCallCache[cacheKey]) {
+          warning(
+            false,
+            'You are manually calling a React.PropTypes validation ' +
+            'function for the prop %s on %s. This is deprecated ' +
+            'and will not work in the next major version.',
```

Comment:
Updated the above message: please include the link. (It doesn’t actually work yet but I’ll fix it tomorrow). We need to explain that [some third-party proptypes](https://github.com/react-bootstrap/react-prop-types/blob/master/src/all.js) might need to be updated to pass `...rest` arguments down now, or they would trigger a false positive.